sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-total-partner-income.e2e-spec.ts

```diff
@@ -0,0 +1,97 @@
+import { PROGRAM_YEAR } from "../../constants/program-year.constants";
+import {
+  ZeebeMockedClient,
+  createFakeConsolidatedPartTimeData,
+  executePartTimeAssessmentForProgramYear,
+} from "../../../test-utils";
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-total-partner-income.`, () => {
+  it("Should calculate total partner income as the CRA verified income value when the student is married and the partner is able to report their financial information.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataRelationshipStatus = "married";
+    assessmentConsolidatedData.studentDataIsYourPartnerAbleToReport = true;
+    assessmentConsolidatedData.partner1CRAReportedIncome = 30000;
+
+    // Act
+    const calculatedAssessment = await executePartTimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    // Calculated total partner income must be 30000.
+    expect(
+      calculatedAssessment.variables.calculatedDataPartner1TotalIncome,
+    ).toBe(30000);
+    // As the student did not report the current year partner income, it must be null.
+    expect(
+      calculatedAssessment.variables.calculatedDataCurrentYearPartnerIncome,
+    ).toBe(null);
+  });
+
+  it(
+    "Should calculate total partner income as self reported partner's current year income when the student is married and need to self report the partner financial information" +
+      " as their partner's current year income needs to be reported.",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataRelationshipStatus = "married";
+      assessmentConsolidatedData.studentDataIsYourPartnerAbleToReport = false;
```

Comment:
Just as an observation, the variable `studentDataIsYourPartnerAbleToReport` added during the previous PR may sometimes be misleading IMO.
The partner may be able to report his income but won't because of some other condition.
From the assessment gateway perspective, I would say something like `studentDataWillYourPartnerReportHisIncome` (or something similar) would be precise about the intention and will be clear about "we need to create a supporting user".
I would not consider it a blocker because it may be open for interpretation (or I am just wrong 😉 ).